### ISSUE TITLE
Anomaly "grounding a non evar-free term"

### ISSUE THREAD

**Issue by emmarion** [coq/coq#13154]
The following source code:

```
Inductive PHolder : Prop -> Prop :=
| pholder : forall P : Prop, PHolder P.

Axiom ax : forall P : Prop, True.

Definition foo (H : PHolder True) : I = I.
  refine 
  (let sofar := match H in PHolder A_ return
    match ax A_ with
    | I => I
    end = I
  with
  | pholder _ => _
  end in _).
  destruct (ax _).
```

Generates the following error:

```
<in exception printer>:<original exception:Anomaly "grounding a non evar-free term"
Please report at http://coq.inria.fr/bugs/.
```

#### Coq Version

CoqIDE 8.12.0

**Comment by SkySkimmer** [comment]
In master it's a regular error.

**Comment by herbelin** [comment]
Probably one of the cases fixed by #13108 then, which is reassuring.
